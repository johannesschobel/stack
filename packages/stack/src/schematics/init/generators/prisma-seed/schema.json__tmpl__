{
  "cli": "nx",
  "id": "prisma-seed",
  "type": "object",
  "properties": {
    "confirm": {
      "type": "boolean",
      "description": "Library confirm",
      "x-prompt": "Are you sure you want to delete and seed the database?"
    }
  },
  "required": ["confirm"]
}
