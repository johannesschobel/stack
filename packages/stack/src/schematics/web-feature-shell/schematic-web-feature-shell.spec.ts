import { Tree } from '@angular-devkit/schematics'
import { SchematicTestRunner } from '@angular-devkit/schematics/testing'
import { createEmptyWorkspace } from '@nrwl/workspace/testing'
import { join } from 'path'

import { WebFeatureShellSchematicSchema } from './schema'

describe('web-feature-shell schematic', () => {
  let appTree: Tree
  const options: WebFeatureShellSchematicSchema = { appName: 'test' }

  const testRunner = new SchematicTestRunner('@nxpm/web-feature-shell', join(__dirname, '../../../collection.json'))

  beforeEach(() => {
    appTree = createEmptyWorkspace(Tree.empty())
  })

  it('should run successfully', async () => {
    await expect(
      testRunner.runSchematicAsync('web-feature-shell', options, appTree).toPromise(),
    ).resolves.not.toThrowError()

    expect(appTree.read('libs/test/shell/feature/src/lib/test-shell-feature.module.ts').toString()).toMatchSnapshot()
  })
})
