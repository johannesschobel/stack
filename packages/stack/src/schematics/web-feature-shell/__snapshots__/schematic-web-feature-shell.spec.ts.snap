// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`web-feature-shell schematic should run successfully 1`] = `
"import { CommonModule } from '@angular/common'
import { NgModule } from '@angular/core'
import { RouterModule, Routes } from '@angular/router'
import { TestAuthDataAccessModule, IsLoggedInGuard } from '@nxpm/test/auth/data-access'
import { TestCoreDataAccessModule } from '@nxpm/test/core/data-access'
import { TestLayoutComponent } from '@nxpm/test/layout'

const routes: Routes = [
  {
    path: '',
    component: TestLayoutComponent,
    canActivate: [IsLoggedInGuard],
    children: [
      // Application routes here
      { path: '', pathMatch: 'full', redirectTo: 'shell' },
      {
        path: 'about',
        loadChildren: () => import('@nxpm/test/about/feature').then((m) => m.TestAboutFeatureModule),
      },
      {
        path: 'account',
        loadChildren: () => import('@nxpm/test/account/feature').then((m) => m.TestAccountFeatureModule),
      },
      {
        path: 'admin',
        loadChildren: () => import('@nxpm/test/admin/feature').then((m) => m.TestAdminFeatureModule),
      },
      {
        path: 'shell',
        loadChildren: () => import('@nxpm/test/shell/feature').then((m) => m.TestShellFeatureModule),
      },
      { path: 'not-found', loadChildren: () => import('./not-found/not-found.module').then((m) => m.NotFoundModule) },
    ],
  },
  {
    path: '',
    loadChildren: () => import('@nxpm/test/auth/feature').then((m) => m.TestAuthFeatureModule),
  },
  { path: '**', redirectTo: '/not-found' },
]

@NgModule({
  imports: [CommonModule, RouterModule.forRoot(routes), TestCoreDataAccessModule, TestAuthDataAccessModule],
})
export class TestShellFeatureModule {}
"
`;
